tests/shell/tests.py

```diff
@@ -29,3 +39,25 @@ def test_stdin_read(self, select):
             stdin.seek(0)
             call_command('shell')
         self.assertEqual(stdout.getvalue().strip(), '100')
+
+    # If we don't patch select then the following tests will fail when the test
+    # suite is run in parallel mode.  This is because the the tests are run in
+    # a subprocess and the subprocess's stdin is closed and replaced by
+    # /dev/null.  Reading from /dev/null always returns EOF and so select will
+    # always show that sys.stdin is ready to read.  This causes problems
+    # because of the call to select.select towards the end of
+    # django.core.management.commands.shell.BaseCommand.handle.
+
+    @unittest.skipIf(IPython is not None, 'IPython is installed')
```

Comment:
Smart... This only works because `shell.py` does inner imports, but that does seem unlikely to be refactored